base/src/main/java/org/aya/states/TyckState.java

```diff
@@ -114,7 +117,36 @@ public void solveMetas(@NotNull Reporter reporter) {
     }
   }
 
-  public @NotNull Term computeSolution(@NotNull MetaCall meta, @NotNull UnaryOperator<Term> f) {
+  public @NotNull Term computeSolution(
+    @Closed @NotNull MetaCall meta,
+    @Bound @NotNull MetaVar.OfType.ClassType classType,
+    @NotNull UnaryOperator<@Closed Term> f
+  ) {
+    var insted = classType.instTele(meta.args().view());
+    insted.instances().filter(it -> {
+      return switch (it) {
+        // try replace all param of [def] with meta, then solve by unify with [insted.type()]
+        case Instance.Global(var def) -> throw new UnsupportedOperationException("TODO");
+        case Instance.Local(var ref, var ty) -> {
+          // TODO: which unifier we should use?
+          // i guess this won't cause infinite recursion, as the context of `meta` doesn't contain itself
+          Unifier someUnifier = null;
+          var result = someUnifier.compare(insted.type(), ty, null);
+          // still keep unsure
+          // TODO: what if YES for 1 instance and UNSURE for many other instances, is it possible?
```

Comment:
It will be possible, and I think we should keep all the unsure instances in that case.